src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/AdjournHearingsWindowMappingTest.java

```diff
@@ -0,0 +1,187 @@
+package uk.gov.hmcts.reform.sscs.helper.mapping;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@TestInstance(PER_CLASS)
+class AdjournHearingsWindowMappingTest {
+    private static final String DATE_TO_BE_FIXED = "dateToBeFixed";
+    private static final String FIRST_AVAILABLE_DATE = "firstAvailableDate";
+    private static final String FIRST_AVAILABLE_DATE_AFTER = "firstAvailableDateAfter";
+    private static final String PROVIDE_PERIOD = "providePeriod";
+    private static final String PROVIDE_DATE = "provideDate";
+    private static final int DAYS_TO_ADD_HEARING_WINDOW_TODAY_ADJOURNMENT = 14;
+
+    @DisplayName("When case is adjourned with date type 'date to be fixed' "
+        + "returns null date and saves 'date to be fixed' to 'Other' field")
+    @Test
+    void whenCaseIsAdjournedWithDateTypeToBeFixed_returnsNullDate_andSavesDateToBeFixedInOtherField() {
+        assertReturnsExpectedWithDateType(DATE_TO_BE_FIXED, null);
```

Comment:
Personally, I think extracting the test logic out into a separate method makes it harder to read for not much gain

Reply:
done